pypeit/scripts/coadd_2dspec.py

```diff
@@ -110,8 +114,13 @@ def main(args):
 
     head2d = fits.getheader(spec2d_files[0])
     if args.basename is None:
-        filename = os.path.basename(spec2d_files[0])
-        basename = filename.split('_')[2]
+        #TODO Fix this, currently does not work if target names have - or _
```

Comment:
`rsplit` and using the `maxsplit` kwarg for split might help here so you only split by 1 -/_ delimiter at a time

Reply:
Thanks, this needs to be revamped from scratch.